File: solution/1491. Average Salary Excluding the Minimum and Maximum Salary/1491.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        int mx = INT_MIN, mn = INT_MAX;
        double sum = 0;
        for (int x : salary) {
            mx = max(mx, x);
            mn = min(mn, x);
            sum += x;
        }
        return (sum - mx - mn) / (salary.size() - 2);
    }
};

File: solution/1491. Average Salary Excluding the Minimum and Maximum Salary/1491.py
class Solution:
    def average(self, salary: List[int]) -> float:
        total, mn, mx = 0.0, math.inf, -math.inf
        for x in salary:
            mn, mx = min(mn, x), max(mx, x)
            total += x
        return (total - mn - mx) / (len(salary) - 2)

